### PR TITLE
Update look of the website

### PR DESCRIPTION
**Pull request by evfinkn**
- Creates a main layout containing a navigation bar and footer.
  - The navbar has links to the home page, products page, each category (all of which currently go nowhere), and the login page. When the user is logged in, the "Login" button is instead a dropdown containing a link to their profile and to logout. There's also a search bar (which currently does nothing), and the active page's link in the navbar is brighter than the other links.
  - The footer contains links to the home page as well as dead links to the features page, FAQs page, and about page.
- Styles the login page.
  - This includes adding a link to the signup page under the "Login" button. Additionally, a "Remember me" checkbox and "Forgot password?" link have been added, but they currently do nothing.
- Styles the signup page.
  - This includes adding a link to the login page under the "Signup" button.
- Styles the flash.
  - Besides a new look, there's now a dismiss button.

In order to do all of this styling, some refactoring had to be done:

- The `current_user` attribute of `ApplicationController` was moved to the new `Current` class so that it could be accessed from views.
- Setting `Current.user` and redirecting to the login page when not logged-in are now two different methods.
- The `importmap-rails` was added gem in order to use Bootstrap's JavaScript functions, enabling dropdowns to function properly.
- The `turbo-rails` and `stimulus-rails` gems were added to speed up the website.
- The `inline_svg` gem was added along with three SVG icons for use in flash messages.
- Helpers were made to separate most of the logic from the views as well as to make elements reusable.

Here is what the updated pages look like:

styled-home-page-with-flash
styled-login-page
styled-signup-page

### CODE REVIEW

**Comment by powersaudrey25** [comment]
Looks awesome! The checkout page will be hosted at "/checkout", we should probably put that in the navbar

File: app/assets/stylesheets/application.scss [hunk 1]
```diff
@@ -15,16 +15,18 @@
...
 }
 
-.flash-alert {
-    background-color: #FF5733;
-    color: #fff;
-    padding: 10px;
-    text-align: center;
+.top-15 {
+    top: 15% !important;
 }
```
Comment (blhohmann) [inline app/assets/stylesheets/application.scss:32]:
```suggestion
}
.dropdown-menu {
    display: none;
}

.dropdown:hover .dropdown-menu {
    display: block;
}
```

**Review by blhohmann — CHANGES_REQUESTED** [review]
Looks really professional, phenomenal work Evan.

The refactors also will save us lots of time, good touch.

I added a suggestion for readding hovering to the dropdowns because I personally like hovering to show dropdown contents on the navbar. If you removed it because you think the hover wouldn't be a good fit that's fine, but I think it improves the usability of the website. Let me know what you think.

File: app/assets/stylesheets/application.scss (hunk 1 above)
Comment (evfinkn) [reply app/assets/stylesheets/application.scss:32]: I didn't realize I removed it, good catch!

**Review by blhohmann — APPROVED** [review]

**Pull request merged**